src/containers/profile/ProfileSections/AccountSettings/AccountSettings.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import "./AccountSettings.css";
+import { connect } from "react-redux";
+import ProfileInfo from "./AccountSettingsSections/ProfileInfo/ProfileInfo";
+import UpdateEmail from "./AccountSettingsSections/UpdateEmail";
+import UpdatePassword from "./AccountSettingsSections/UpdatePassword";
+import DeleteAccount from "./AccountSettingsSections/DeleteAccount";
+
+const AccountSettings = (props) => {
+  const { id, auth } = props;
+
+  return (
+    <div className="accountSettings">
+      <ProfileInfo id={id} auth={auth} />
+      {auth.uid === id && (
```

Comment:
why does the id in the props need to be checked against auth.uid? is someone else looking at someone else's auth settings?

Reply:
yes, this condition will prevent the admins from accessing a user's personal page (either the user was a subscriber, author, or another admin)

Reply:
ok. just so you know, this is fine for now but in a real app we can't do it like this, the check needs to be serverside. (you should look at firebase security rules -- they allow you to do this).

the reason is that we have access to javascript code in the client side, so i could load the site, delete this if-condition, and access this anyway